### ISSUE TITLE
Extension to Estimator API: Output shape different from prediction shape

### ISSUE THREAD

**Issue by mandar2812** [eaplatanios/tensorflow_scala#82]
I am experimenting with some self implemented loss functions like [`RBFWeightedSWLoss`](https://github.com/transcendent-ai-labs/PlasmaML/blob/master/helios/src/main/scala-2.11/io/github/mandar2812/PlasmaML/helios/core/RBFWeightedSWLoss.scala).

## Problem Definition

The key idea behind these loss functions is that the shape of the output prediction might not match the shape of the output labels.

In this particular example `RBFWeightedSWLoss`, the output is consists of a 2 dimensional vector which has 
  1 target prediction `f`
  2 time index when it will be observed. `i`

The targets consist of sliding time windows of the target signal `y(t)`

The loss is then computed as a kernel weighted average as specified in the source.

## State of Affairs

Currently, the `Estimator` API throws an exception in this case

```
java.lang.IllegalArgumentException: Expected output shapes compatible with '([?, 128, 128, 4],[?])', but got dataset with output shapes '([?, 128, 128, 4],[?, 72])'.
  org.platanios.tensorflow.api.ops.io.data.Iterator.createInitializer(Iterator.scala:67)
  org.platanios.tensorflow.api.learn.estimators.FileBasedEstimator$$anonfun$trainWithHooks$1.apply$mcV$sp(FileBasedEstimator.scala:155)
  org.platanios.tensorflow.api.learn.estimators.FileBasedEstimator$$anonfun$trainWithHooks$1.apply(FileBasedEstimator.scala:137)
  org.platanios.tensorflow.api.learn.estimators.FileBasedEstimator$$anonfun$trainWithHooks$1.apply(FileBasedEstimator.scala:137)

```

## Proposal

I am wondering if it is possible to have a new feature/workaround of this problem. It would be super helpful if its not necessary for the output and predictions to match in shape as long as the loss can compute the loss function given the two.

**Comment by eaplatanios** [comment]
@mandar2812 No worries! :) I think the issue lies with your `trainInput`. The dataset you're loading seems to load tensors of shape `[-1, 72]` as the labels for supervision, where you define your train input as:

```scala
val trainInput = tf.learn.Input(FLOAT32, Shape(-1))
```

Try changing that to the following and see if the inputs to the model work:

```scala
val trainInput = tf.learn.Input(FLOAT32, Shape(-1, 72))
```

I hope this helps. :)

**Comment by mandar2812** [comment]
@eaplatanios Its working now. Thanks a lot!
